process/timeout.rs

```diff
@@ -0,0 +1,283 @@
+#[command(author, version, about, long_about)]
+struct Args {
+    /// Only time out the utility itself, not its descendants.
+    #[arg(short = 'f', long)]
+    foreground: bool,
+
+    /// Always preserve (mimic) the wait status of the executed utility, even if the time limit was reached.
+    #[arg(short = 'p', long)]
+    preserve_status: bool,
+
+    /// Send a SIGKILL signal if the child process created to execute the utility has not terminated after the time period
+    /// specified by time has elapsed since the first signal was sent. The value of time shall be interpreted as specified for
+    /// the duration operand.
+    #[arg(short = 'k', long, value_parser = parse_duration)]
+    kill_after: Option<Duration>,
+
+    /// Specify the signal to send when the time limit is reached, using one of the symbolic names defined in the <signal.h> header.
+    /// Values of signal_name shall be recognized in a case-independent fashion, without the SIG prefix. By default, SIGTERM shall be sent.
+    #[arg(short = 's', long, default_value = "TERM", value_parser = parse_signal)]
+    signal: Signal,
+
+    /// The maximum amount of time to allow the utility to run, specified as a decimal number with an optional decimal fraction and an optional suffix.
+    #[arg(name = "DURATION", value_parser = parse_duration)]
+    duration: Duration,
+
+    /// The name of a utility that is to be executed.
+    #[arg(name = "UTILITY")]
+    utility: String,
+
+    /// Any string to be supplied as an argument when executing the utility named by the utility operand.
+    #[arg(name = "ARGUMENT", trailing_var_arg = true)]
+    arguments: Vec<String>,
+}
+
+/// Parses string slice into [Duration].
+///
+/// # Arguments
+///
+/// * s - [str] that represents input duration.
+///
+/// # Errors
+///
+/// Returns an error if passed invalid input.
+///
+/// # Returns
+///
+/// Returns [Duration].
+fn parse_duration(s: &str) -> Result<Duration, String> {
+    let (value, suffix) = s.split_at(
+        s.find(|c: char| !c.is_ascii_digit() && c != '.')
+            .unwrap_or(s.len()),
+    );
+
+    let value: f64 = value
+        .parse()
+        .map_err(|_| format!("invalid duration format '{}'", s))?;
+
+    let multiplier = match suffix {
+        "s" | "" => 1.0,
+        "m" => 60.0,
+        "h" => 3600.0,
+        "d" => 86400.0,
+        _ => return Err(format!("invalid duration format '{}'", s)),
+    };
+
+    Ok(Duration::from_secs_f64(value * multiplier))
+}
+
+/// Parses and validates the signal name, returning its integer value.
+///
+/// # Arguments
+///
+/// * s - [str] that represents the signal name.
+///
+/// # Errors
+///
+/// Returns an error if passed invalid input.
+///
+/// # Returns
+///
+/// Returns the integer value of the signal.
+fn parse_signal(s: &str) -> Result<Signal, String> {
+    let signal_name = format!("SIG{}", s.to_uppercase());
+
+    Signal::from_str(&signal_name).map_err(|_| format!("invalid signal name '{}'", s))
+}
```

Comment:
In the original utility, input with the SIG suffix is supported. Please fix it.

Reply:
Fixed